lib/course_planner_web/controllers/auth/password_controller.ex

```diff
@@ -8,12 +8,40 @@ defmodule CoursePlannerWeb.Auth.PasswordController do
 
   alias CoursePlanner.Accounts.{Users, User}
   alias CoursePlannerWeb.{Auth.UserEmail, Router.Helpers}
+  alias Recaptcha.Config
 
   def new(conn, _) do
-    render conn, "new.html"
+    render(conn, "new.html", errors: [])
   end
 
-  def create(conn, %{"password" => %{"email" => email}}) do
+  def create(conn, %{"password" => create_params, "g-recaptcha-response" => recaptcha_response}) do
+    case Recaptcha.verify(recaptcha_response) do
+      {:ok, _response} ->
+        conn
+        |> do_create(%{"password" => create_params})
+
+      {:error, _errors} ->
+        errors = [recaptcha: {"Captcha is not validated", []}]
+
+        conn
+        |> render("new.html", errors: errors)
+    end
+  end
+  def create(conn, %{"password" => %{"email" => _email}} = params) do
+    recaptcha_noconfigured? =
+      is_nil(Config.get_env(:recaptcha, :secret))
+        or is_nil(Config.get_env(:recaptcha, :public_key))
```

Comment:
well, it provide the option of not having recaptcha. so in case it is not on the server it will return as nil. (E.g. no heroku variable set for it ) and therefore application will work without it.
it isn't good ? should we enforce it ?

Reply:
That would be better as a setting, not as a "config existing or not".
It's confusing for the user and a little big bug prone.

Reply:
okay, but it is not user itself, it is the person that setup the server.
anyhow if recaptcha is not configured, shouldn't we still let the application to work ?

Reply:
Yeah, but there's a difference between the admin disabling it explicitly in the settings and forgetting to setup the configs. If he forgets, we should give an error and he either configures it or disables it. We shouldn't let them continue using the software without making that decision I believe.

Reply:
but if admin wants to disable it first should be able to login. for that the config should be correct first. do you get my point ? i think it is a bit circular

Reply:
In that case, the setting should be disabled by default.
I still disagree, but the alternatives I'm giving aren't too different or better in hindsight, so I'll approve the way it is.

Reply:
shall we deal with this as a design discussion under another card ?
for now we can go on with this and I create a card for it

Reply:
That'll be cool